src/api/ledgerManager.js

```diff
@@ -67,6 +67,33 @@ export class LedgerManager {
     );
   }
 
+  /**
+   * Persists the local (in-memory) Ledger to the ledger storage. Reloads the
+   * remote ledger from storage right before persisting it to minimize the
+   * possibility of overwriting remote changes that were not synced to the local
+   * ledger and ensure consistency of the ledger events (e.g. no double spends).
+   */
+  async persist(): Promise<ReloadResult> {
+    let reloadRes = await this.reloadLedger();
+    if (reloadRes.error) {
+      return reloadRes;
+    }
+
+    // START RACE CONDITION
+    await this._storage.write(this._ledger);
+    // END RACE CONDITION
```

Comment:
It says the race condition starts and ends here, but doesn't it actually start on line 77 when we reload the ledger?

Suppose you have client A using this code. If client B writes any time between when the ledger was reloaded and when it gets written, it potentially triggers conflicts, yeah?

Reply:
Maybe write more detailed comments explaining what kind of race condition specifically exists on line 83?

Reply:
Yea, you're right, that's how I intended to communicate it. Race condition starts after the ledger is reloaded, so I put the "START RACE CONDITION" comment after the reloadLedger command, but putting it before would be easier to understand. Will update!

Reply:
Added this comment in the code, reposting here for context:
```
   * A race condition is present in this function: if client A runs reloadLedger
   * and then client B writes to the remote ledger before client A finishes writing,
   * then the changes to the ledger that client B made would be overwritten by the
   * changes from client A. The correctness and consistency of the ledger will still
   * be maintained, its just that client B might experience data loss of whatever
   * events they were trying to sync. To detect if this has occurred, we reload
   * the ledger again after writing the data to ensure the local changes were
   * not overwritten. If they were, we can show an error message to client B with
   * a list of changes that failed to sync.
```